### ISSUE TITLE
Look into caching ZPT documents

### ISSUE THREAD

**Issue by craigfowler** [csf-dev/ZPT-Sharp#36]
It might be possible to improve performance by caching the documents loaded from disk. This would require a custom document factory and template factory, but might improve performance.

**Comment by craigfowler** [comment]
This applies to the MVC view engine.

**Comment by craigfowler** [comment]
We would no longer load them afresh on each call (such as when they are requested as a view, or as a way if getting a macro). Instead we would cache the document instance in-memory, indexed by its file path, so that subsequent calls return the cached instance.

**Comment by craigfowler** [comment]
This would be implemented at the document factory level. A subclass of the factory would be used which also caches the loaded documents.

We would need a cache provider also.

**Comment by craigfowler** [comment]
Deferred until after #98

**Comment by craigfowler** [comment]
Closing as invalid - it's before it's time.  We really don't know where the biggest performance drains are right now.  Once we do then this may well come back in another form.
